src/rgw/rgw_sal_motr.cc

```diff
@@ -2990,6 +3023,20 @@ int MotrStore::store_access_key(const DoutPrefixProvider *dpp, optional_yield y,
   return rc;
 }
 
+int MotrStore::store_email_key(const DoutPrefixProvider *dpp, optional_yield y, MotrEmailInfo emailInfo)
+{
+  int rc;
+  bufferlist bl;
+  emailInfo.encode(bl);
+  rc = do_idx_op_by_name(RGW_IAM_MOTR_EMAIL_KEY,
+                                M0_IC_PUT, emailInfo.email_id, bl);
+  if (rc < 0){
+    ldout(cctx, 0) << "Failed to store key: rc = " << rc << dendl;
```

Comment:
Please update key -> email

Reply:
Didnt understand. Have passed email_id as key only

Reply:
Key here represents access key. Error should indicate what it failed to store.

Reply:
Not changing this

Reply:
Fixed